fault-proof/docker-compose.yml

```diff
@@ -13,10 +13,51 @@ services:
   challenger:
     build:
       context: ..
-      dockerfile: fault_proof/Dockerfile.challenger
+      dockerfile: fault-proof/Dockerfile.challenger
     image: op-succinct-fp-challenger:latest
     container_name: op-succinct-fp-challenger
     restart: unless-stopped
     env_file:
       - .env.challenger
     command: ["challenger"]
+
+  prometheus:
+    image: prom/prometheus:latest
+    container_name: prometheus
+    restart: unless-stopped
+    ports:
+      - "${FP_PROMETHEUS_PORT:-9090}:9090"
+    volumes:
+      - ./prometheus:/etc/prometheus
+      - prometheus_data:/prometheus
+      - ./.env.proposer:/tmp/.env.proposer
+      - ./.env.challenger:/tmp/.env.challenger
+    user: root
+    entrypoint: >
+      /bin/sh -c "
+        PROPOSER_PORT=$(grep PROPOSER_METRICS_PORT /tmp/.env.proposer | cut -d= -f2) &&
+        CHALLENGER_PORT=$(grep CHALLENGER_METRICS_PORT /tmp/.env.challenger | cut -d= -f2) &&
+        sed -i \"s/proposer:9000/proposer:$${PROPOSER_PORT}/g\" /etc/prometheus/prometheus.yml &&
+        sed -i \"s/challenger:9001/challenger:$${CHALLENGER_PORT}/g\" /etc/prometheus/prometheus.yml &&
+        exec /bin/prometheus --config.file=/etc/prometheus/prometheus.yml --storage.tsdb.path=/prometheus --web.console.libraries=/etc/prometheus/console_libraries --web.console.templates=/etc/prometheus/consoles --web.enable-lifecycle
+      "
+
+  grafana:
+    image: grafana/grafana:latest
+    container_name: grafana
+    restart: unless-stopped
+    ports:
+      - "${FP_GRAFANA_PORT:-3000}:3000"
+    volumes:
+      - ./grafana/provisioning:/etc/grafana/provisioning
+      - ./grafana/dashboards:/var/lib/grafana/dashboards
+      - ./grafana/alerts:/var/lib/grafana/alerts
+      - grafana_data:/var/lib/grafana
+    environment:
+      - GF_SECURITY_ADMIN_USER=admin
+      - GF_SECURITY_ADMIN_PASSWORD=admin
+      - GF_USERS_ALLOW_SIGN_UP=false
+
+volumes:
+  prometheus_data:
+  grafana_data:
```

Comment:
This feels overly engineered. Simplify and just have it work in the default case.